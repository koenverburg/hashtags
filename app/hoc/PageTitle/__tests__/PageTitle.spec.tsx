import React from 'react'
import { JSDOM } from 'jsdom'
// tslint:disable-next-line:no-implicit-dependencies
import { ShallowWrapper, shallow } from 'enzyme'
import { WithPageTitle } from '../index'

const MockComponent = props => <p>test</p>

describe('HOC - WithPageTitle', () => {
  let wrapper: ShallowWrapper

  beforeAll(() => {
    const { window } = new JSDOM('<!doctype html><html><body></body></html>')
    // @ts-ignore
    global.window = window
    // @ts-ignore
    global.document = window.document

    const Test = WithPageTitle('Mock page title')(MockComponent)
    wrapper = shallow(<Test />)
  })

  it('will renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // describe('it should mark the component name as wrapped with', () => {
    // it.todo("should use 'displayName' for getDisplayName method", () => {
    //   expect(wrapper.instance.displayName).toBeUndefined()
    // })

    // it.todo("should use 'name' if 'displayName' is not available for the getDisplayName method", () => {
    //   expect(wrapper.instance.name).toEqual('WithPageTitle(MockComponent)')
    // })

    // tslint:disable-next-line:max-line-length
    // it("should fallback to 'Component' if 'displayName' and 'name' aren't available on the wrappedComponent", () => {
    //   // expect(wrapper.name()).toEqual('WithPageTitle(MockComponent)')
    //   expect(wrapper.displayName).toEqual('WithPageTitle(MockComponent)')
    // })
  // })
})
