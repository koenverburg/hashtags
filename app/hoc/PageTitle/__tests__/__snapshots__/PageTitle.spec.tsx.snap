// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC - WithPageTitle will renders 1`] = `<MockComponent />`;
